### ISSUE TITLE
`avoid_unnecessary_return_variable` triggers when type promotion is needed

### ISSUE THREAD

**Issue by andrew-bekhiet-solid** [solid-software/solid_lints#226]
```dart
T get<T>(String key) {
  final value = _map[key];
  
  if (value is T) {
    // local variable is promoted to T
    return value;
  }

  throw Exception('value is not of type $T');
}
```

### Expected behavior
No lint

### Actual behavior
The lint rule is triggered

**Comment by solid-illiaaihistov** [comment]
Fixed in: https://github.com/solid-software/solid_lints/pull/292

**Comment by solid-illiaaihistov** [comment]
Fixed
